scrapy_crawlera/spidermiddlewares.py

```diff
@@ -0,0 +1,42 @@
+from scrapy import Request
+
+
+class CrawleraSessionReuseMiddleware(object):
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        return cls(crawler)
+
+    def __init__(self, crawler):
+        setting = 'CRAWLERA_SESSION_REUSE_DEFAULT_SESSION'
+        self._default_session = crawler.settings.get(setting)
+
+    def process_spider_output(self, response, result, spider):
+        def _set_session(request_or_item):
+            if not isinstance(request_or_item, Request):
+                return request_or_item
+
+            request = request_or_item
+            header = b'X-Crawlera-Session'
+
+            if request.headers.get(header) != b'_reuse':
```

Comment:
I don't know how I feel about giving "made-up" `X-Crawlera-Session` values. I just think about the users that also use Crawlera directly, as this could be misleading. Maybe it is just the developer in me thinking this way, I guess there is no a lot of users that will get confused with this, as they maybe use only scrapy or only crawlera directly.

Maybe using a `meta` argument instead?

Reply:
`meta={'crawlera_session_reuse': True}`?

Reply:
yeah, that looks good, of course that modification changes more parts of this implementation right? 😅

Reply:
It’s not that bad, I’ll update it tomorrow.